File: 0067-add-binary/0067-add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        c=0
        if not a or not b:
            return a+b
        n1 = len(a)-1
        n2 = len(b)-1
        if n1>n2:
            b = "0"*(n1-n2)+b
        else:
            a="0"*(n2-n1)+a
        rs = ''
        n1=len(a)-1
        while n1>=0:
            v = (int(a[n1])+int(b[n1]))+c
            if v==2:
                c=1
                v=0
            elif v==3:
                c=1
                v=1
            else:
                c=0
            rs = str(v)+rs
            n1-=1
            if c>0 and (n1==-1):
                rs= str(c)+rs
        return rs

File: 0067-add-binary/0067-add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        int c=0;
        StringBuilder sb = new StringBuilder();
        int i =  a.length()-1;
        int j = b.length()-1;
        while(i>=0 || j>=0 || c==1){
            if (i>=0){
                c=c+ a.charAt(i)- '0'; //here a.charAt(i) = 49 if the value was 1 else 48 if it was 0 as char is converted to ascii since we try to do 0+'1'-'0'= 0+49-1
                i-=1;
            }
            if (j>=0){
                c=c+b.charAt(j)-'0'; //same as above
                j-=1;
            }
            sb.append(c%2);
            c=c/2;
        }
        return sb.reverse().toString();
    }
}